File: plugins/pi/src/TDPIConnector/TDPIConnector.Web/src/app/models/monitoring-info.ts
import { PIEventReceived } from "./pi-event-received";
import { AFEventReceived } from "./af-event-received";
import { EventsPerPoint } from "./events-per-point";
import { EventsPerAttribute } from "./events-per-attribute";
import { ResponsesPerCode } from "./responses-per-code";
import { PIConnection } from "./pi-connection";
import { TDEngineHttpResponseSummary } from "./td-engine-http-response-summary";
import { TDEngineInfo } from "./td-engine-info";

export interface MonitoringInfo 
{
    piConnection: PIConnection
    eventsPerPoint: EventsPerPoint[] 
    eventsPerAttribute: EventsPerAttribute[] 
    tdEngineHttpResponses:  TDEngineHttpResponseSummary[]
    responsesPerCode: ResponsesPerCode[]
    tdEngineInfo: TDEngineInfo
    lastPIEvents: PIEventReceived[] 
    lastAFEvents: AFEventReceived[] 
}

File: source/taos-xservice/plugins/pi/src/TDPIConnector/TDPIConnector.Core/Monitoring/Models/MonitoringInfo.cs
using System.Collections.Generic;
using TDPIConnector.PI;
using TDPIConnector.TDEngine.Models;

namespace TDPIConnector.Core.Monitoring.Models
{
    public class MonitoringInfo
    {
        public PIConnection PIConnection { get; internal set; }
        public List<EventsPerPoint> EventsPerPoint { get; internal set; }
        public List<EventsPerAttribute> EventsPerAttribute { get; internal set; }
        public List<TDEngineHttpResponseSummary> TDEngineHttpResponses { get; internal set; }
        public List<PIEventReceived> LastPIEvents { get; internal set; }
        public List<AFEventReceived> LastAFEvents { get; internal set; }
        public List<ResponsesPerCode> ResponsesPerCode { get; set; }
        public TDEngineInfo TDEngineInfo { get; internal set; }
   
    }
}
